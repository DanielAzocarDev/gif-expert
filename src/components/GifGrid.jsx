import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      <div className="card-grid">
        {loading ? (
          <p>Cargando...</p>
        ) : (
          data.map((gif) => <GifGridItem {...gif} key={gif.id} />)
        )}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
