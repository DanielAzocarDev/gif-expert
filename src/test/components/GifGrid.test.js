
import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock('../../hooks/useFetchGifs')

describe('test <GifGrid />', () => {
  
  const category = 'Goku'
  

  test('should render component', () => {
    const gifs = []
    useFetchGifs.mockReturnValue({data: gifs, loading: true});
    const wrapper = shallow(<GifGrid category={category}/>)

    expect(wrapper).toMatchSnapshot()
  })

  test('should show components', () => {
    const gifs = [{
      id: 12,
      title: 'hola',
      url: 'https//:www.hola.com',
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow(<GifGrid category={category}/>)
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })
})