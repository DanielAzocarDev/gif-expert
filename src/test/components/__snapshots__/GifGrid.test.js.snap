// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifGrid /> should render component 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <p>
      Cargando...
    </p>
  </div>
</Fragment>
`;
